### PR TITLE
Auth Provider methods with AuthUI parameter

### PR DESCRIPTION
**Pull request by wimbledon**
#973

### CODE REVIEW

**Comment by wimbledon** [comment]
In addition to using the default FUIAuth's defaultAuthUI, it would be nice to pass in a custom FUIAuth to create auth provider.

**Comment by wimbledon** [comment]
@morganchen12, Would you mind taking a look at this PR?

File: FirebaseOAuthUI/Sources/FUIOAuth.m [hunk 1]
```diff
@@ -196,11 +213,19 @@ + (FUIOAuth *)yahooAuthProvider {
...
   UIUserInterfaceStyle style = UITraitCollection.currentTraitCollection.userInterfaceStyle;
-  return [self appleAuthProviderWithUserInterfaceStyle:style];
+  return [self appleAuthProviderWithAuthUI:authUI userInterfaceStyle:style];
 }
 
 + (FUIOAuth *)appleAuthProviderWithUserInterfaceStyle:(UIUserInterfaceStyle)userInterfaceStyle {
+    return [FUIOAuth appleAuthProviderWithAuthUI:[FUIAuth defaultAuthUI] userInterfaceStyle:userInterfaceStyle];
+}
+
++ (FUIOAuth *)appleAuthProviderWithAuthUI:(FUIAuth *)authUI userInterfaceStyle:(UIUserInterfaceStyle)userInterfaceStyle {
```
Comment (morganchen12) [inline FirebaseOAuthUI/Sources/FUIOAuth.m:228]: nit: Line break after the first parameter

**Review by morganchen12 — COMMENTED** [review]
Had one nit, but that can be addressed after merging. Thanks @wimbledon!

File: FirebaseOAuthUI/Sources/FUIOAuth.m (hunk 1 above)
Comment (wimbledon) [reply FirebaseOAuthUI/Sources/FUIOAuth.m]: Fixed

**Comment by wimbledon** [comment]
@morganchen12, fixed the nit.
would you mind merging this PR?
I don't think I have permission to merge yet.

**Review by morganchen12 — APPROVED** [review]

**Pull request merged**